### DISCUSSION TITLE
How does `useGroups` works for `getMarkerFeatures`

### DISCUSSION THREAD

**Discussion by Zepeng-Mu** [GreenleafLab/ArchR#1854] [Questions / Documentation] [answered]
Hello,
I'm trying to do a pairwise differential test in `getMarkerFeatures`. From the document, the purpose of `useGroups` is `A character vector that is used to select a subset of groups by name from the designated groupBy column in cellColData. This limits the groups used to perform marker feature identification.` Therefore I assume that if I pass a vector to it, all elements in the vector will be grouped together and treated the same. However, when I run the function in this way
```r
useGroups = 1:10%&%"Disease",
bgdGroups = 1:10%&%"Healthy",
```
The results I got looks like this:
```r
class: SummarizedExperiment 
dim: 462 10 
metadata(2): MatchInfo Params
assays(7): Log2FC Mean ... AUC MeanBGD
rownames(462): 1 2 ... 461 462
rowData names(6): seqnames start ... name idx
colnames(10): 1Disease 2Disease ... 9Disease 10Disease
colData names(0):
```
which means that each element in `useGroups` is treated separately. But this effect does not apply to `bgdGroups`.
```r
useGroups = "1Disease",
bgdGroups = 1:10%&%"Healthy",
```
gives
```r
class: SummarizedExperiment 
dim: 31824 1 
metadata(2): MatchInfo Params
assays(7): Log2FC Mean ... AUC MeanBGD
rownames(31824): 1 2 ... 31823 31824
rowData names(6): seqnames start ... name idx
colnames(1): 1Disease
colData names(0):
```
I just wonder if this is the expected behavior of the function?
In any case, I can just collapse all the labels by myself before running differential test.

Thanks!

**Comment by rcorces** [comment, marked as answer]
Are you worried because there is an individual column for each group in the `SummarizedExperiment` output? This is the expected behavior. At run time, `useGroups` and `bgdGroups` are used to find matched sets of cells to compare. But ArchR still reports the LogFC per group and attempts to find marker features per group. The way you are trying to use this function is analytically indistinguishable from the use case of finding marker features across multiple groups. If you want to perform a single pairwise test, then you should make a new column in `cellColData` and define all cells that correspond to your two groups of interest and then pass one of those groups to `useGroups` and the other to `bgdGroups`. That is the safest way to use that function.

**Reply by Zepeng-Mu** [reply]
Thanks! This is really helpful.
